appinventor/blocklyeditor/src/generators/yail.js

```diff
@@ -114,6 +114,18 @@ Blockly.Yail.FLONUM_REGEXP = "^[\\s]*[-+]?([0-9]*)((\\.[0-9]+)|[0-9]\\.)[\\s]*$"
  * @returns {String} the generated code if there were no errors.
  */
 Blockly.Yail.getFormYail = function(formJson, packageName, forRepl, workspace) {
+  var oldForRepl = this.forRepl;
+  var code;
+  try {
+    this.forRepl = forRepl;
+    code = Blockly.Yail.getFormYail_(formJson, packageName, forRepl, workspace);
+  } finally {
+    this.forRepl = oldForRepl;
```

Comment:
I can't figure out where/how this is being used, but it's probably just because I've been gone so long. I'm going to trust you on it :D

Reply:
This was mostly for clarity. The `getFormYail_` method is the old definition of the function. This code sets the `forRepl` property and resets it to whatever it used to be, even in the case of a failure to generate the code (via the `try-finally`). When we are generating code for the build server (i.e., not the REPL), we don't need to include the `protect-enum` call since we know the buildserver has the latest functionality.

Reply:
See the patch to `blocklyeditor/src/yail/helpers.js` for where this is now used.

Reply:
Ohhh gotcha @ewpatton I was looking for somewhere it was used within the file, but I forgot it was public! Logic makes a lot more sense now, lgtm =)

Reply:
Ok. Feel free to merge if you're happy with it. Once the PR is merged into App Inventor this code will be under your name.